assets/src/stories-editor/components/higher-order/with-context-menu.js

```diff
@@ -56,49 +58,64 @@ const applyWithSelect = withSelect( ( select, props ) => {
 		if ( toolBar ) {
 			toolBarHeight = toolBar.clientHeight;
 		}
-		const relativePositionX = event.clientX - wrapperDimensions.left;
-		const relativePositionY = event.clientY - wrapperDimensions.top - toolBarHeight;
+
+		let eventX = 0;
+		let eventY = 0;
+
+		if ( isRightClick ) {
+			// Use coordinates of actual click to place context menu.
+			eventX = event.clientX;
+			eventY = event.clientY;
+		} else {
+			// Place menu in the upper left corner of the element - but slightly inside.
```

Comment:
Would it perhaps make sense to use the center of the element instead?

Reply:
Asking because the current behavior looks odd for rotated blocks.

Reply:
Great suggestion, I will check the possibilities.

Reply:
Added some Suggested to that works on the center point of element.